pika/adapters/base_connection.py

```diff
@@ -382,27 +365,27 @@ def _handle_events(self, fd, events, error=None, write_only=False):
             self._handle_write()
             self._manage_event_state()
 
-        if self.socket and not write_only and (events & self.READ):
+        if self.socket and (events & self.READ):
             self._handle_read()
 
-        if (self.socket and write_only and (events & self.READ) and
-            (events & self.ERROR)):
-            error_msg = ('BAD libc:  Write-Only but Read+Error. '
-                         'Assume socket disconnected.')
-            LOGGER.error(error_msg)
-            self._on_terminate(connection.InternalCloseReasons.SOCKET_ERROR,
-                               error_msg)
 
         if self.socket and (events & self.ERROR):
-            LOGGER.error('Error event %r, %r', events, error)
-            self._handle_error(error)
+            # No need to handle the ERROR event separately from I/O. If there is
+            # a stream error, READ and/or WRITE will also be inidicated and the
+            # corresponding _handle_write/_handle_read will get tripped up with
+            # the approrpriate error and terminate the connection. So, we'll
+            # just debug-log the information in case it helps with diagnostics.
+            error = self.socket.getsockopt(socket.SOL_SOCKET, socket.SO_ERROR)
+            LOGGER.debug('Sock error events %r; sock error code %r', events,
+                         error)
 
     def _handle_read(self):
         """Read from the socket and call our on_data_available with the data."""
         try:
             while True:
                 try:
                     if self.params.ssl:
+                        # TODO Why using read instead of recv on ssl socket?
```

Comment:
I suspect it is due to [this bug](https://bugs.python.org/issue3890) which has been resolved for a long time.